aristanetworks/ansible-avd
# Feat(eos_cli_config_gen): Add ECN Support

## Change Summary

Added support for ECN configuration in uc-tx-queues and tx-queues is qos profiles.
Added support for ECN count command on the ethernet interfaces. This is required on some platforms where the count command on the qos profiles does not apply to the interfaces.



## Related Issue(s)

No issues raised.
- This PR will add to mainline some features that are in use by customers right now by means of custom jinja templates

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes



- Update the `eos_cli_config_gen` schema to support `random-detect ecn` for tx-queues and uc-tx-queues
- Update template to generate the configuration for tx-queues and uc-tx-queues
- Update templates to allow the use of tx-queue and uc-tx-queue commands within the ethernet port

## How to test



- Run the molecule scenario `eos_cli_config_gen`. The `qos.cfg` should include the commands to configure ecn on tx and uc-tx queues as well as an example setting `ecn count` on an ethernet interface.
- 
## Checklist

### User Checklist


- Verify that the generated config does not have typos
- Verify that the way the schema is defined makes sense for someone using ECN for the first time
- Verify that the Documentation is clear enough.

### Repository Checklist
